packages/fetchai/connections/oef/connection.py

```diff
@@ -421,6 +460,7 @@ def send_oef_message(self, envelope: Envelope) -> None:
                 "Could not create dialogue for message={}".format(oef_message)
             )  # pragma: nocover
             return  # pragma: nocover
+        oef_search_dialogue.envelope_context = envelope.context
```

Comment:
We use the same context on response. If a skill_id is set in the context it gets applied by AEA when routing the envelope.